src/main/java/hudson/plugins/ec2/EC2OndemandSlave.java

```diff
@@ -109,6 +111,7 @@ public void terminate() {
                             Jenkins.get().removeNode(this);
                             LOGGER.info("Removed EC2 instance from jenkins controller: " + getInstanceId());
                         } catch (AmazonClientException | IOException e) {
+                            listener.error("Failed to terminate EC2 instance: " + getInstanceId());
                             LOGGER.log(Level.WARNING, "Failed to terminate EC2 instance: " + getInstanceId(), e);
```

Comment:
Could perhaps remove the system log message and send the stack trace to the `listener` (see `Functions.printStackTrace`).

Reply:
fixed in 8b95a323909b18a7bf4bba82fe92ad942520dfec